### PR TITLE
Add SSR support for cart

### PR DESCRIPTION
**Pull request by andrzejewsky**
- [x] I Accept [Individual Contributor License Agreement](https://github.com/DivanteLtd/next/blob/master/CLA.md)

### CODE REVIEW

File: packages/core/factories/src/useCartFactory.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { UseCart } from '@vue-storefront/interfaces';
-import { Ref, ref, computed } from '@vue/composition-api';
+import { Ref, ref, computed, isRef } from '@vue/composition-api';
```
Comment (Qrzy) [inline packages/core/factories/src/useCartFactory.ts:2]: I can't see any usage of `isRef`. Is it necessary here?

File: packages/core/factories/src/useCartFactory.ts [hunk 2]
```diff
@@ -72,7 +76,11 @@ export function useCartFactory<CART, CART_ITEM, PRODUCT, COUPON>(
     };
 
     const refreshCart = async () => {
-      factoryParams.cart.value = await factoryParams.loadCart();
+      loading.value = true;
+      const cart = await factoryParams.loadCart();
```
Comment (Qrzy) [inline packages/core/factories/src/useCartFactory.ts:80]: Shouldn't this be wrapped with try-catch-finally?

**Review by Qrzy — APPROVED** [review]

File: packages/core/factories/src/useCartFactory.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { UseCart } from '@vue-storefront/interfaces';
-import { Ref, ref, computed } from '@vue/composition-api';
+import { Ref, ref, computed, isRef } from '@vue/composition-api';
+import { useSSR } from '@vue-storefront/utils';
 
 export type UseCartFactoryParams<CART, CART_ITEM, PRODUCT, COUPON> = {
```
Comment (filrak) [inline packages/core/factories/src/useCartFactory.ts:5]: would be nice to have an option to turn SSR off OOTB

File: packages/core/theme-module/theme/components/checkout/CartPreview.vue [hunk 4]
```diff
@@ -121,14 +122,18 @@ export default {
...
+    const { cart, removeFromCart, updateQuantity, refreshCart } = useCart();
     const listIsHidden = ref(false);
     const promoCode = ref('');
     const showPromoCode = ref(false);
     const products = computed(() => cartGetters.getItems(cart.value));
     const totalItems = computed(() => cartGetters.getTotalItems(cart.value));
     const totals = computed(() => cartGetters.getTotals(cart.value));
 
+    onSSR(async () => {
+      await refreshCart();
```
Comment (filrak) [inline packages/core/theme-module/theme/components/checkout/CartPreview.vue:134]: why do we need to refresh cart on checkout?

File: packages/core/theme-module/theme/components/CartSidebar.vue [hunk 5]
```diff
@@ -93,11 +95,15 @@ export default {
...
   },
   setup() {
-    const { cart, removeFromCart, updateQuantity } = useCart();
+    const { cart, removeFromCart, updateQuantity, refreshCart } = useCart();
     const products = computed(() => cartGetters.getItems(cart.value));
     const totals = computed(() => cartGetters.getTotals(cart.value));
     const totalItems = computed(() => cartGetters.getTotalItems(cart.value));
 
+    onSSR(async () => {
+      await refreshCart();
```
Comment (filrak) [inline packages/core/theme-module/theme/components/CartSidebar.vue:104]: why do we need to do a cart refresh?

**Comment by andrzejewsky** [comment]
https://github.com/DivanteLtd/next/pull/329

**Pull request closed without merging**